bin/hdfcoinc/pycbc_coinc_statmap

```diff
@@ -133,20 +242,313 @@ logging.info("calculating ifar/fap values")
+
+    # Repopulate the list of foreground indices for the loudest events 
+    loud_t_ind = []
+    for i in range(0, len(fnlouder)):
+        if fnlouder[i] == 0:
+            loud_t_ind.append(i)
+
+# End of the while loop, now take all of the information and write it back in
+# to file the way that other codes downstream know how to read.
+logging.info("Writing final calculations to file")
+
+for key in all_trigs.seg.keys():
+    f['segments/%s/start' % key] = f['segments_h_iter0/%s/start' % key]
+    f['segments/%s/end' % key] = f['segments_h_iter0/%s/end' % key]
+
+# If there were hierarchically removed foreground triggers, please  write
+# them back into the file with the correct statistics.
+if len(save_these_fore_ind) > 0:
+#   CONTROVERSIAL: Which one do I write here? It won't match if I write the background
+#   back for the last iteration. I don't know what to use.
+    f['segments/foreground_veto/start'] = f['segments_h_iter0/foreground_veto/start']
+    f['segments/foreground_veto/end'] =  f['segments_h_iter0/foreground_veto/end']
+    for k in copy_trigs.data:
+        f['foreground/' + k] = f['foreground_h_iter0/' + k]
+ 
+    # We won't run into trouble inserting the foreground triggers back in
+    # if we insert them in from least to greatest. Otherwise we could really
+    # mess up the ordering of assigning the proper ifar to every foreground
+    # trigger. So we need to preserve the original ordering!
+    sort_like_this = numpy.argsort(save_these_fore_ind)
+    save_these_fore_ind = save_these_fore_ind[sort_like_this]
+    save_ifar_fore_trig = save_ifar_fore_trig[sort_like_this]
+    save_fap_fore_trig = save_fap_fore_trig[sort_like_this]
+
+    # The indices of hierarchically removed indices will shift around as
+    # we put them back into the list of foreground triggers. So save the
+    # final foreground trigger indices for writing to file.
+    final_fore_ind = []
+
+    for i in range(0, len(save_these_fore_ind)):
+        # Handle the edge case that you have to add to the end of the list
+        # if your array of ifar has gotten too small. Also save final
+        # foreground indices for writing to file.
+        if save_these_fore_ind[i] > len(ifar):
+            ifar = np.append(ifar, save_ifar_fore_trig[i])
+            final_fore_ind.append(save_ifar_fore_trig[i])
+        else :
+            ifar = np.insert(ifar, save_these_fore_ind[i] + i, save_ifar_fore_trig[i])
+            final_fore_ind.append(save_these_fore_ind[i] + i)
+
+        if save_these_fore_ind[i] > len(fap):
+            fap = np.append(fap, save_fap_fore_trig[i])
+        else :
+            fap = np.insert(fap, save_these_fore_ind[i] + i, save_fap_fore_trig[i])
+
+    f['foreground/ifar'] = sec_to_year(ifar)
+    f['foreground/fap'] = fap
+
+    # Write the foreground indices that have been hierarchical removed for
+    # programs downstream.
+    f['hierarchically-removed-fore-indices'] = np.array(final_fore_ind[:])
```

Comment:
Everything uses underscores and here you're using dashes. Might want to change this to underscores.

Also other things you've named start ``hierarchical_*`` not ``hierarchically_*``.